server/container/guice/jpa-common-guice/src/main/java/org/apache/james/modules/data/JPAEntityManagerModule.java

```diff
@@ -49,15 +49,20 @@ public EntityManagerFactory provideEntityManagerFactory(JPAConfiguration jpaConf
         
         properties.put("openjpa.ConnectionDriverName", jpaConfiguration.getDriverName());
         properties.put("openjpa.ConnectionURL", jpaConfiguration.getDriverURL());
+        jpaConfiguration.getCredential()
+            .ifPresent(credential -> {
+                properties.put("openjpa.ConnectionUserName", credential.getUsername());
+                properties.put("openjpa.ConnectionPassword", credential.getPassword());
+            });
 
         List<String> connectionFactoryProperties = new ArrayList<>();
         connectionFactoryProperties.add("TestOnBorrow=" + jpaConfiguration.isTestOnBorrow());
-        if (jpaConfiguration.getValidationQueryTimeoutSec() > 0) {
-            connectionFactoryProperties.add("ValidationTimeout=" + jpaConfiguration.getValidationQueryTimeoutSec() * 1000);
-        }
-        if (jpaConfiguration.getValidationQuery() != null) {
-            connectionFactoryProperties.add("ValidationSQL='" + jpaConfiguration.getValidationQuery() + "'");
-        }
+        jpaConfiguration.getValidationQueryTimeoutSec()
+            .filter(timeoutSecond -> timeoutSecond > 0)
```

Comment:
so 0 and negative numbers are the same as empty. Could we forbid these values in the Configuration ?